src/useScript/useScript.ts

```diff
@@ -1,67 +1,115 @@
 import { useEffect, useState } from 'react'
 
-export type Status = 'idle' | 'loading' | 'ready' | 'error'
-export type ScriptElt = HTMLScriptElement | null
+export type UseScriptStatus = 'idle' | 'loading' | 'ok' | 'error'
+export interface UseScriptOptions {
+  shouldLoad?: boolean
+  removeOnUnmount?: boolean
+}
+
+const defaultOptions: UseScriptOptions = {
+  shouldLoad: true,
+  removeOnUnmount: true,
+}
```

Comment:
You wouldn't need these if you swap `shouldLoad` to `shouldPreventLoad` and `removeOnUnmount` actually removes the `scriptNode` itself.